tt_metal/impl/buffers/buffer.cpp

```diff
@@ -28,7 +28,11 @@ void validate_buffer_size_and_page_size(
     const BufferType &buffer_type,
     const TensorMemoryLayout &buffer_layout,
     const std::optional<ShardSpecBuffer>& shard_parameters) {
-    TT_FATAL(size != 0 and page_size != 0, "Buffer size and page size should be larger than 0 bytes!");
+    TT_FATAL(page_size != 0, "Buffer page size should be larger than 0 bytes!");
```

Comment:
Should be 0 too, I think I saw an assert before for this

Reply:
@abhullar-tt @ayerofieiev-tt  Actually page_size must be any non-zero number in this case. There are a lot of place in the code where we divide `size` (or something like size) by `page_size`, which would fail if `page_size` is zero. This is why there's this TT_FATAL in constructor.

Reply:
Update: now the page_size for 0-size buffer can be arbitrary, including 0